scripts/ghutils.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+# This script contains functions for interacting with github.
+# At one stage it contained code for setting github statuses,
+# but that is currently handled by the jenkins github pull requests
+# builder plugin, so the code was removed from here.
+
+
+import json
+
+import click
+import requests
+
+
+def _add_token(url, pat):
+    url = "{url}?access_token={pat}".format(url=url, pat=pat)
+    return url
+
+
+@click.command()
+@click.option('--tag', help='Jenkins build tag')
+@click.option('--link', help='Link to related build in Jenkins UI')
+@click.option('--org', help='Github Organisation that owns the target repo')
+@click.option('--repo', help='Name of target repo')
+@click.option('--pat', help="Github Personal Access Token")
+@click.option('--label',
+              help="Add label to issue, can be spcified multiple times",
+              multiple=True)
+def create_issue(tag, link, org, repo, pat, label):
+    issue_create_url = _add_token(
+        "https://api.github.com/repos/{org}/{repo}/issues".format(
```

Comment:
Python modules exist for interacting with GitHub, e.g github3.py. I think they simplify the code.

```python
import github3

gh = github3.login(token=pat)
issue = {
    "title": "JBF: {tag}".format(tag=tag),
    "body": "[link to failing build]({url})".format(url=link),
    "labels": label
}

repository = gh.repository(org, repo)
repository.create_issue(**issue)
```